tests/Detector.Tests/Go/GoDetectorTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Detector.Go;
+using Microsoft.Oryx.Tests.Common;
+using Xunit;
+namespace Microsoft.Oryx.Detector.Tests.Go
+{
+    public class GoDetectorTest : IClassFixture<TestTempDirTestFixture>
+    {
+
+        private const string GoFileWithVersion = @"
+            module hello
+
+            go 1.16
+        ";
```

Comment:
I think it would be worth having a test where the version specified isn't in the format `MAJOR.MINOR` (maybe missing a minor version or has a patch version, ensuring we're handling it correctly)

Reply:
Will take a deeper look at what is a valid version for go and adjust accordingly 👍